core/src/main/scala/kafka/coordinator/transaction/TransactionMetadata.scala

```diff
@@ -135,7 +135,7 @@ private[transaction] class TransactionMetadata(val producerId: Long,
   }
 
   def removePartition(topicPartition: TopicPartition): Unit = {
-    if (pendingState.isDefined || (state != PrepareCommit && state != PrepareAbort))
+    if (pendingState.isDefined && (state != PrepareCommit && state != PrepareAbort))
```

Comment:
Hmm, I think it is not expected if `state != PrepareCommit && state != PrepareAbort` itself, what state it was when hitting this exception?